app/assets/javascripts/notifications.js

```diff
@@ -0,0 +1,106 @@
+
+      $(".actions").on("click", function (e) {
+        var elem = $(e.target),
+            notificationFeed = $(".notification-feed"),
+            userAction;
+
+        if (elem.data("accept")) {
+          userAction = $.ajax({
+            url: ("/dashboard/notifications/" + elem.data("accept")),
+            method: "PUT",
+            data: { status: "active" }
+          }).done(function () {
+            swal({
+              title: "Task accepted",
+              text: "You can view all your assigned tasks from the manage tasks page",
+              type: "success",
+              confirmButtonColor: "#eb4d5c"
+            });
+
+            displayContext.empty().append("<p>Click on the view button to get more information about a task</p>")
+              .append("<i class='material-icons'>directions_bike</i>");
+
+            $(".feed .btn[data-id=" + elem.data("accept") + "]").closest(".feed").css("opacity", 0);
+
+            window.setTimeout(function () {
+              $(".feed .btn[data-id=" + elem.data("accept") + "]").closest(".feed").remove();
+            }, 1000);
+
+            if (notificationFeed.children().length === 1) {
+              displayContext.empty().append("<p>No new notifications available</p>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>");
+            }
+          });
+        } else {
+          userAction = $.ajax({
+            url: ("/dashboard/notifications/" + elem.data("reject")),
+            method: "PUT",
+            data: { status: "rejected" }
+          }).done(function () {
+            swal({
+              title: "Task rejected",
+              text: "The tasker will be notified of your choice",
+              type: "error",
+              confirmButtonColor: "#eb4d5c"
+            });
+
+            displayContext.empty().append("<p>Click on the view button to get more information about a task</p>")
+              .append("<i class='material-icons'>directions_bike</i>");
+
+            $(".feed .btn[data-id=" + elem.data("reject") + "]").closest(".feed").css("opacity", 0);
+
+            window.setTimeout(function () {
+              $(".feed .btn[data-id=" + elem.data("reject") + "]").closest(".feed").remove();
+            }, 1000);
+
+            if (notificationFeed.children().length === 1) {
+              displayContext.empty().append("<p>No new notifications available</p>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>")
+              .append("<i class='material-icons'>highlight_off</i>");
+            }
+          });
+        }
+      });
+    });
```

Comment:
Almost all the code in the if block is duplicated in the else block. Do you think it will make sense to `DRY` things up?